packages/@aws-cdk/aws-cloudformation/test/deps.test.ts

```diff
@@ -0,0 +1,339 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { ResourcePart } from '@aws-cdk/assert-internal';
+import '@aws-cdk/assert-internal/jest';
+import { App, CfnResource, Stack } from '@aws-cdk/core';
+import * as cxapi from '@aws-cdk/cx-api';
+import { NestedStack } from '../lib';
+
+describe('resource dependencies', () => {
+  test('between two resources in a top-level stack', () => {
+    // GIVEN
+    const app = new App();
+    const stack = new Stack(app, 'Stack');
+    const r1 = new CfnResource(stack, 'r1', { type: 'r1' });
+    const r2 = new CfnResource(stack, 'r2', { type: 'r2' });
+
+    // WHEN
+    r1.addDependsOn(r2);
+
+    // THEN
+    expect(app.synth().getStackArtifact(stack.artifactId).template).toEqual({
+      Resources:
+        { r1: { Type: 'r1', DependsOn: ['r2'] }, r2: { Type: 'r2' } },
+    });
+  });
+
+  // eslint-disable-next-line jest/valid-describe
+  describe('resource in nested stack depends on a resource in the parent stack', matrixForResourceDependencyTest((addDep) => {
```

Comment:
[jest/valid-describe](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/valid-describe.md). It appears that the `describe` callback function should not contain any return statement or parameters. I am not actually sure which is the error here as `matrixForResourceDependencyTest` does return `() => {}`. However, the actual function does have both a `return` statement and parameters, which may be what is causing the error.

At any rate, I think that this is a valid way to utilize `describe` with as much fidelity as possible, and the underlying tests work as expected. Just want a second pair of eyes here.